addons/sale/tests/test_credit_limit.py

```diff
@@ -63,3 +75,26 @@ def test_credit_limit_multicurrency(self):
             "partner_a has reached its Credit Limit of : $\xa050.00\n"
             "Total amount due (including this document) : $\xa055.00"
         )
+
+    @users('notaccountman')
+    def test_credit_limit_access(self):
+        """Ensure credit warning gets displayed without Accounting access."""
+        for group in self.partner_a._fields['credit'].groups.split(','):
+            self.assertFalse(self.env.user.has_group(group))
+
+        self.partner_a.sudo().credit_limit = self.product_a.list_price
+
+        with Form(self.empty_order.with_env(self.env)) as order_form:
```

Comment:
The test didn't fail even when removing the `sudo` on `order_sudo`, because `@users` only changes `self.env`, not `self.empty_order.env`.  Pushed a little fixup commit to fix that ;)

Reply:
Good catch, kinda assumed it wouldn't fail yet anyway till the 18.0 forward port.

Reply:
Actually, the test failed in that case because I forgot to set `sale.order.user_id` to the current user, leading to the access error.

Accessing `credit` or `credit_limit` without accounting privileges doesn't result in an access error in 16.0 (yet).

Reply:
Oh yeah, didn't investigate much, but wanted to make sure the test was covering what we wanted correctly :D
The explanation makes sense, thanks :+1: